AmplifyPlugins/Auth/AWSAuthPlugin/Dependency/AuthorizationProviderAdapter+SignedInSession.swift

```diff
@@ -0,0 +1,194 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+import AWSCore
+import AWSMobileClient
+import AWSPluginsCore
+
+extension AuthorizationProviderAdapter {
+
+    /// Creates session for a signedIn user.
+    ///
+    /// Session values available as a signedIn user depend on the way the user signedIn and the configurations enabled
+    /// for the current signedIn user. The following scenarios can happen:
+    /// - User is signedIn through Cognito User Pool and Identity Pool is not configured
+    /// - User is signedIn through Cognito User Pool but refresh token is expired
+    /// - User is signedIn through Cognito User Pool but access token couldnot be refreshed because of N/W or
+    /// service issue
+    /// - User is signedIn through Cognito User Pool but aws credentials couldnot be refreshed because of N/W or
+    /// service issue
+    /// - User is signedIn through Cognito Identity Pool
+    ///- User is signedIn through Cognito Identity Pool but aws credentials couldnot be refreshed because of N/W or
+    ///service issue
+    /// - Parameter completionHandler: Completion handler to return the result.
+    func fetchSignedInSession( _ completionHandler: @escaping (Result<AuthSession, AmplifyAuthError>) -> Void) {
+
+        // User can be signed in to user pools or federated to identityPool.
+        // Call getTokens first to verify that the signedIn user is in UserPool or not.
```

Comment:
Note this as an explicit precondition in the doc comments, that this must be called only after verifying that the user is not in `guest` or `signedOut` states, and that the operation is happening in a thread-safe way

Reply:
Added